### PR TITLE
Feature/nav 76 application properties spring with environment variables logging dependencies constants etc

### PR DESCRIPTION
**Pull request by munterfi**
PR introduces new `application.properties`:

- Systematic naming of properties, also in configuration.
- Adjust service config, parser and initialiser to new names.
- Replace meters per hour with SI unit meters per second for walking.
- Use cron instead of hours to define the update interval.
- Avoid nested parallel streams in generators.
- Add search radius as further parameter (rename maxWalkDistance).
- File downloader and repo / infrastructure layer.

@clukas1 Can you please review. I hope all parameters are still used at the correct locations. 

Maybe especially check: 
- `transfer.time.same.stop.default` vs. `transfer.time.between.stops.minimum`
- `walking.duration.minimum` and `walking.search.radius`

A further issue, which is not entirely clear to me, is the distinction between same stop transfer and same station transfer. Are stations meant to be the parent stop with all children stops? So is same station and same stop transfer time not the same?

If they are the same, and there seems to be no influence in the generators on children parent relations in the generated transfers, then maybe we could rename it all to same stop.

### CODE REVIEW

File: src/main/java/ch/naviqore/app/infrastructure/GtfsScheduleUrl.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+public class GtfsScheduleUrl implements GtfsScheduleRepository {
+
+    private final String url;
+    private final String downloadDirectory;
+    private final String fileName;
+
+    @Override
+    public GtfsSchedule get() throws IOException, InterruptedException {
+        Path directory = Path.of(downloadDirectory);
+        new FileDownloader(url).downloadTo(directory, fileName, true);
```
Comment (clukas1) [inline src/main/java/ch/naviqore/app/infrastructure/GtfsScheduleUrl.java]: Maybe add check if file already exists (and is up to date, given a specified update frequency) before downloading?

File: src/main/java/ch/naviqore/app/infrastructure/GtfsScheduleUrl.java [hunk 2]
```diff
@@ -7,19 +7,27 @@
...
     private final String url;
-    private final String downloadDirectory;
-    private final String fileName;
 
     @Override
     public GtfsSchedule get() throws IOException, InterruptedException {
-        Path directory = Path.of(downloadDirectory);
-        new FileDownloader(url).downloadTo(directory, fileName, true);
-        return new GtfsScheduleReader().read(directory.resolve(fileName).toString());
+        Path tempDir = Files.createTempDirectory(TMP_DIRECTORY_PREFIX);
```
Comment (clukas1) [inline src/main/java/ch/naviqore/app/infrastructure/GtfsScheduleUrl.java:23]: also an option :)

**Review by clukas1 — COMMENTED** [review]
Nice work cleaning up after me. Don't see any required changes, but maybe best if we just go through together and discuss all application properties one by one before finalizing this.

File: src/main/java/ch/naviqore/app/infrastructure/GtfsScheduleUrl.java (hunk 1 above)
Comment (munterfi) [reply src/main/java/ch/naviqore/app/infrastructure/GtfsScheduleUrl.java]: Good point, i created another issue for this, NAV-102

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,12 +1,47 @@
...
\ No newline at end of file
+# Walking speed in meters per second. The default value is based on the average preferred walking speed.
+# Source: https://en.wikipedia.org/wiki/Preferred_walking_speed
+walking.speed=${WALKING_SPEED:1.4}
+# Maximum walking distance in meters. No walks longer than this will appear in connections (first/last mile and
+# between station transfers).
+walking.distance.maximum=${WALKING_DISTANCE_MAXIMUM:500}
+# Minimum walking duration in seconds. If walk duration is shorter, no first or last mile walk is needed to reach the
+# location. Avoids very short walks in stations that feign a false sense of accuracy.
+walking.duration.minimum=${WALKING_DURATION_MINIMUM:120}
```
[the remark this answers is not in the document]
Comment (munterfi) [reply src/main/resources/application.properties]: Discusses in weekly.

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1,12 +1,47 @@
...
+# WALKING
+# ==============================================
+# Type of walking calculator used to calculate walking distances. Currently only BEE_LINE_DISTANCE.
 walking.calculator.type=${WALKING_CALCULATOR_TYPE:BEE_LINE_DISTANCE}
-walking.speed=${WALKING_SPEED:2500}
-walking.distance.maximum=${WALKING_DISTANCE_MAXIMUM:500}
\ No newline at end of file
+# Walking speed in meters per second. The default value is based on the average preferred walking speed.
+# Source: https://en.wikipedia.org/wiki/Preferred_walking_speed
+walking.speed=${WALKING_SPEED:1.4}
```
[the remark this answers is not in the document]
Comment (munterfi) [reply src/main/resources/application.properties:42]: see comment below

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1,12 +1,47 @@
...
+# Cron expression for updating the static GTFS feed from the provided URL. Public transit agencies update their static
+# GTFS data regularly. Set this interval to match the agency's publish schedule. Default is to update the schedule
+# daily at 4 AM.
+gtfs.static.update.cron=${GTFS_STATIC_UPDATE_CRON:0 0 4 * * *}
+# ==============================================
+# TRANSFER
+# ==============================================
+# Minimum transfer time between the same stops in seconds. If stops are closer than this, this time has precedence over
+# the actual walking time, which accounts for leaving the station building, stairways, etc.
+transfer.time.between.stops.minimum=${TRANSFER_TIME_BETWEEN_STOPS_MINIMUM:120}
```
[the remark this answers is not in the document]
Comment (munterfi) [reply src/main/resources/application.properties]: Introduced in [246bbc0](https://github.com/naviqore/public-transit-service/pull/62/commits/246bbc0b2e01eb14da4da32063b650a4d2ef12a1)

**Comment by munterfi** [comment]
Changes:

- Add access and egress time twice to walk transfers between different stops.
- Add access or egress time once to first and last mile walks.
- Consistently rename stations to stops.

In my notion, a transit stop is a designated location where passengers can access or egress from a transit vehicle, typically with minimal infrastructure. In contrast, a transit station is a larger, more complex facility that serves as a central hub for multiple transit routes or services, with extensive infrastructure.

The term transit station could also be used in the future to reflect parent stops that have child stops.

Before working on different branches i would suggest merging this or branchin from here, since the renaming is quite accross the project (should have done this in a separate PR, sorry...).

**Comment by munterfi** [comment]
image

I think we still need to tune the default parameters, since instead of driving to the main station with tram 3, our router suggests egressing a stop earlier at Löwenplatz and walking to the station, which is something no one would do. 😄

**Comment by clukas1** [comment]
> image
> 
> I think we still need to tune the default parameters, since instead of driving to the main station with tram 3, our router suggests egressing a stop earlier at Löwenplatz and walking to the station, which is something no one would do. 😄

Absolutely, that's why I thought we should introduce more "costs" to walks.

**Review by clukas1 — APPROVED** [review]
Looks good

**Pull request merged**